core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -670,10 +683,7 @@ class Partition(val topicPartition: TopicPartition,
             // update ISR in ZK and cache
             expandIsr(newInSyncReplicaIds)
           }
-          // check if the HW of the partition can now be incremented
-          // since the replica may already be in the ISR and its LEO has just incremented
-          maybeIncrementLeaderHW(leaderLog, followerFetchTimeMs)
```

Comment:
I think this is a good change. The intuition is that it is impossible for the high watermark to advance by adding new members to the set. The previous value reflected the smallest end offset among the current replicas in the ISR and the new replica is guaranteed to be at or above that. Whatever the case, the smallest end offset would remain the same.